src/libraries/Common/src/System/Security/Cryptography/Helpers.cs

```diff
@@ -45,12 +45,12 @@ internal static partial class Helpers
         [return: NotNullIfNotNull(nameof(src))]
         public static byte[]? CloneByteArray(this byte[]? src)
         {
-            if (src == null)
+            return src switch
             {
-                return null;
-            }
-
-            return (byte[])(src.Clone());
+                null => null,
+                { Length: 0 } => src,
+                _ => (byte[])src.Clone(),
+            };
```

Comment:
FWIW, it could also have been:
```C#
return src is not null ? src.AsSpan().ToArray() : null;
```
and let span's ToArray handle the empty case via Array.Empty.